### PR TITLE
Make graph a value rather than point in CRD types

### LINKED ISSUE #1742: failed calling webhook
when i type command _kubectl apply -f deployment.yaml_, error occurred:

<pre>
Error from server (InternalError): error when creating "/Users/jason/sodb/deployment/deployment.yaml": Internal error occurred: failed calling webhook "v1.mseldondeployment.kb.io": Post https://seldon-webhook-service.seldon-system.svc:443/mutate-machinelearning-seldon-io-v1-seldondeployment?timeout=30s: stream error: stream ID 3; INTERNAL_ERROR
</pre>

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes #1742 

 * Changes the graph to value as it is required.
 * Adds unit test and e2e test

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Apr 27 11:52:58 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=1

**Comment by seldondev** [comment]
Mon Apr 27 11:52:59 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=2

**Comment by seldondev** [comment]
Mon Apr 27 11:55:23 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=4

**Comment by seldondev** [comment]
Mon Apr 27 11:55:29 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=3

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 900e3243bcfd24a15c5d6cac41638760c6ebbe5f | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go [hunk 1]
```diff
@@ -353,6 +352,8 @@ func (r *SeldonDeploymentSpec) DefaultSeldonDeployment(mldepName string, namespa
 				}
 			}
 		}
+
+		r.Predictors[i] = p
```
Comment (adriangonz) [inline operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:356]: Do we need this line?

Comment (cliveseldon) [reply operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:356]: seems we do to get the predictor updated .Good GoLang question as to why...

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1752#issuecomment-625151910" title="Approved">adriangonz</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [adriangonz]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Thu May  7 10:14:55 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=6

**Comment by seldondev** [comment]
Thu May  7 10:14:56 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=5

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 80c13efa468e81241101add16969dbeb49c7f500 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon May 18 09:13:07 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=7

**Comment by seldondev** [comment]
Mon May 18 09:13:17 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=8

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 52a756a0e78594191e222671f98bb7c88ea47570 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon May 18 09:42:49 UTC 2020
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=9

**Comment by seldondev** [comment]
Mon May 18 09:42:57 UTC 2020
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1752/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1752 --build=10

**Pull request merged**